lib/graphql/query/arguments.rb

```diff
@@ -12,8 +12,10 @@ def self.construct_arguments_class(argument_owner)
         argument_owner.arguments_class = Class.new(self) do
           self.argument_definitions = argument_definitions
 
-          def initialize(values)
-            super(values, argument_definitions: self.class.argument_definitions)
+          def initialize(values, values_used_default = Hash.new(false))
```

Comment:
The main reason for this default value is this call: https://github.com/chrisbutcher/graphql-ruby/blob/b9a6401218b76b6dd0c96105816050591dd3928b/lib/graphql/query/arguments.rb#L142

I'm not 100% sure if this is the correct behaviour here.

My initial approach was to pass through the `values_used_default` hash to `super`,  and initialize ArgumentValues with a `default_used` value directly, similar to how `arg_defn` is passed along. Maybe you'd prefer this?

Reply:
I'm honestly not sure what the default value has anything to do with the passing of `values_used_default` to super?

I believe I would prefer that, but I don't see why you'd change it in the first place so I think I'm missing something.

Reply:
I've pushed up the alternate approach here: https://github.com/chrisbutcher/graphql-ruby/pull/1 

Let me know if my description still doesn't clarify things.

Reply:
Personally I prefer that branch

Reply:
Me too, closing this one 🔥